conformance/src/main/scala/za/co/absa/enceladus/conformance/DynamicConformanceJob.scala

```diff
@@ -247,6 +244,23 @@ object DynamicConformanceJob {
     }
   }
 
+  private def handleEmptyOutputAfterConformance(implicit spark: SparkSession): Unit = {
+    val areCountMeasurementsAllZero = Atum.getControMeasure.checkpoints
+      .flatMap(checkpoint =>
+        checkpoint.controls.filter(control =>
+          control.controlName.equalsIgnoreCase(za.co.absa.atum.core.Constants.controlTypeRecordCount)))
+      .forall(m => Try(m.controlValue.toString.toLong).getOrElse(1L) == 0L)
```

Comment:
.getOrElse(1L) == 0L
This could probably be a bit more intuitive ;)

Otherwise interesting idea.

Reply:
I like this approach since it is very short. The alternative is a pattern match against `Success` / `Failure` which might be too verbose for this case.

Reply:
Yeah, this is idea of the one and only (mostly because names on GitHub are unique) @yruslan. But I like it, it is simple and covers all the corners.

Reply:
I agree that it's not the most intuitive thing but it's probably alright.

Reply:
Add a comment?

```
Anything that cannot be converted to Long can be count as a positive count and therefore not 0.
```

Reply:
What about 0.0?

Reply:
```suggestion
      .forall(m => Try(m.controlValue.toString.toLong).toOption.contains(0L))
```

Reply:
Seems like I will need to switch it to double first
```scala
scala> "0.0".toLong
java.lang.NumberFormatException: For input string: "0.0"
  at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
  at java.lang.Long.parseLong(Long.java:589)
  at java.lang.Long.parseLong(Long.java:631)
  at scala.collection.immutable.StringLike$class.toLong(StringLike.scala:277)
  at scala.collection.immutable.StringOps.toLong(StringOps.scala:29)
  ... 51 elided

scala> "0.0".toDouble.toLong
res12: Long = 0
```

Reply:
No need to bother converting to `Long` after that, I think.
```
scala> Option("-0.0".toDouble).contains(0D)
res5: Boolean = true
```

Reply:
I used your option. In the end, we only care about the zero. Did not know about the contains functionality.

Reply:
Can record count even be anything other than a natural number?

Reply:
Control metrics come from source systems, better be defensive I think.

Replying to @benedeki 's question here as well. A control value shouldn't be `null` (never happened yet anyway). If it happens and the job throws a `NullPointerException`, I'd say it is ok. The raw info file needs to be fixed then.

Reply:
I think we can consider this resolved then.